test/test_torch.py

```diff
@@ -16651,16 +16650,155 @@ def test_svd_tall_some_col_maj(self, device, dtype):
+    'log10': ufunc_meta(is_floating=True, complex_on=('cpu', 'cuda')),
+    'log1p': ufunc_meta(is_floating=True, complex_on=()),
+    'log2': ufunc_meta(is_floating=True, complex_on=('cpu', 'cuda')),
+    'sqrt': ufunc_meta(is_floating=True, complex_on=('cpu', 'cuda')),
+    'rsqrt': ufunc_meta(is_floating=True, complex_on=('cpu', 'cuda')),
+    'trunc': ufunc_meta(is_floating=True, complex_on=('cpu', 'cuda')),
+    'erf': ufunc_meta(is_floating=True, complex_on=()),
+    'erfc': ufunc_meta(is_floating=True, complex_on=()),
+    'erfinv': ufunc_meta(is_floating=True, complex_on=()),
+    'lgamma': ufunc_meta(is_floating=True, complex_on=()),
+    'digamma': ufunc_meta(is_floating=True, complex_on=()),
+    'sigmoid': ufunc_meta(is_floating=True, complex_on=('cpu'))
+}
+
+# Creates and adds a test of the given unary floating ufunc's type promotion
+# Unary floating ufuncs promote tensors of bool and integral types to
+# the default scalar type.
+def generate_unary_floating_ufunc_promo_test(cls, op_str):
+    int_types = [torch.bool, torch.uint8, torch.int8, torch.int16,
+                 torch.int32, torch.int64]
+    float_types = [torch.float, torch.double]
+    float_types_cuda = [torch.half, torch.float, torch.double]
+    complex_types = [torch.complex64, torch.complex128]
+    default_types = [torch.float, torch.double]
+    default_types_cuda = [torch.half, torch.float, torch.double]
+
+    def test_fn(self, device):
+        op = getattr(torch, op_str)
+        meth = getattr(torch.Tensor, op_str)
+        my_default_types = default_types
+        my_float_types = float_types
+        if self.device_type == 'cuda':
+            my_default_types = default_types_cuda
+            my_float_types = float_types_cuda
+            # angle_cuda is not implemented for torch.half
+            if op_str == 'angle':
+                my_default_types = [type for type in default_types_cuda if type != torch.half]
+                my_float_types = [type for type in float_types_cuda if type != torch.half]
+        if self.device_type in unary_ufuncs[op_str].complex_on:
+            my_float_types.extend(complex_types)
+
+        prev_def_type = torch.get_default_dtype()
+        try:
+            for in_type, def_type in product(int_types, my_default_types):
+                torch.set_default_dtype(def_type)
+                t = torch.tensor((1,), device=device, dtype=in_type)
+                self.assertEqual(op(t).dtype, def_type)
+                self.assertEqual(meth(t).dtype, def_type)
+        finally:
+            torch.set_default_dtype(prev_def_type)
+
+        for in_type in my_float_types:
+            t = torch.tensor((1,), device=device, dtype=in_type)
+            self.assertEqual(op(t).dtype, in_type)
+            self.assertEqual(meth(t).dtype, in_type)
+
+        # Test for out= code paths
+        for in_type in my_float_types + int_types:
+            if self.device_type not in unary_ufuncs[op_str].complex_on:
```

Comment:
Isn't this break premature? That is, isn't this saying if you don't support complex we should test the out= code path at all?